packages/preview-middleware-client/src/flp/bootstrap.ts

```diff
@@ -0,0 +1,26 @@
+/**
+ * Calculates the script content for accessing the right sap/ushell/bootstrap sandbox.
+ */
+(window as any)['sap-ui-config'] = {
+    'xx-bootTask': ushellBootstrap
+};
+
+function ushellBootstrap(fnCallback: Function) {
+    fetch('/resources/sap-ui-version.json')
```

Comment:
Please use async/await. It makes the code more readable.
Alternatively (and maybe cleaner), use `sap.ui.require` and get `sap/ui/VersionInfo` instead of getting the json file

Reply:
tried this, but it leads to many other eslint problems, fur t the asyn call

Reply:
Hm. It seems to work [here](https://github.com/SAP/open-ux-tools/blob/988727a9a226755df3d90f27b71369597676cd01/packages/preview-middleware-client/src/flp/init.ts#L276) pretty well 🤔 

```ts
import VersionInfo from 'sap/ui/VersionInfo';
//...
export async function ushellBootstrap(fnCallback: () => void)
    const { version } = (await VersionInfo.load()) as { version: string };
    const major = Number(version?.charAt(0)) || 2;
//...
```

or is `'xx-bootTask'` not allowed to return a promise?

Reply:
The problem with the import like that at the top of the page would be executed too early. It can only be executed within the script because only then UI5 is loaded.

Reply:
Ok, understood. But not giving up yet 😄 How about using a dynamic import then?

```ts
//...
export async function ushellBootstrap(fnCallback: () => void)
    const versionInfo = (await import('sap/ui/VersionInfo')).default;
    const { version } = await versionInfo.load() as { version: string };
    const major = Number(version?.charAt(0)) || 2;
//...
```

Reply:
I needed to switch the file to js so that the transpiler doesn't make it UI5 conform. That at least makes the script finally work properly. I will now check your proposal above which looks good to me

Reply:
Fine for me. I don't have anything working right now. It was just an idea. Did not try 🙈 Will push some eslint fixes